### PR TITLE
fix(suite): passphrase missmatch resets the flow now

### PR DESCRIPTION
**Pull request by peter-sanderson**
Addresses the weird behavior when user mismatches the passphrase. Now it resets the flow instead of being stucked in inconsistent state

### CODE REVIEW

File: suite-common/wallet-core/src/device/deviceThunks.ts [hunk 1]
```diff
@@ -428,14 +428,22 @@ export const authConfirm = createThunk(
...
         }
 
         if (response.payload.state !== device.state) {
+            console.log('___authConfirm: ', response.payload.state, '!==', device.state);
+
             dispatch(notificationsActions.addToast({ type: 'auth-confirm-error' }));
-            dispatch(deviceActions.receiveAuthConfirm({ device, success: false }));
+
+            const settings = extra.selectors.selectSuiteSettings(getState());
+            dispatch(deviceActions.forgetDevice({ device, settings }));
```
Comment (peter-sanderson) [inline suite-common/wallet-core/src/device/deviceThunks.ts:445]: :point_up:

File: suite-common/wallet-core/src/discovery/discoveryThunks.ts [hunk 2]
```diff
@@ -407,13 +407,12 @@ export const startDiscoveryThunk = createThunk(
                 await dispatch(initMetadata(false));
             }
 
-            dispatch({
-                type: startDiscovery.type,
```
Comment (peter-sanderson) [inline suite-common/wallet-core/src/discovery/discoveryThunks.ts:411]: this is some old way of dispatching actions and it was hard to find the usages when used this way, lets use actionCreator instead

File: packages/connect-web/src/index.ts [hunk 3]
```diff
@@ -156,8 +156,8 @@ export class TrezorConnectDynamicImpl implements ConnectFactoryDependencies {
         return this.getTarget().requestWebUSBDevice();
     }
 
-    public cancel() {
-        return this.getTarget().cancel();
+    public cancel(error?: string) {
```
Comment (peter-sanderson) [inline packages/connect-web/src/index.ts:159]: cc @martykan :scream:  :grin:

Comment (peter-sanderson) [reply packages/connect-web/src/index.ts:159]: Todo: would be great to have somehow test for this

Comment (martykan) [reply packages/connect-web/src/index.ts:159]: Oops, missed that one. But it's strange TS didn't catch it, the interface should be typed correctly

Comment (peter-sanderson) [reply packages/connect-web/src/index.ts:159]: Its optional, thats probably it. 

I am more surprised that no test had failed.... We have a logic in suite that relies on the passage of errors through this.

Comment (peter-sanderson) [reply packages/connect-web/src/index.ts:159]: Here for example: https://github.com/trezor/trezor-suite/pull/12977/files#diff-2580f7331a6878d62ed30bb9de2c33adbf4566b5d67d8cf33f5705f48af51930R415

(this is how I found it)

**Review by jvaclavik — APPROVED** [review]

**Pull request merged**